### PR TITLE
Containerized proxy services installed in 4.3-nightly and Uyuni PR tests

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Containerized proxy services installed in 4.3 and Uyuni PR tests

### CODE REVIEW

**Comment by srbarrios** [comment]
Testing it here https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests-qe/7/

**Comment by srbarrios** [comment]
> Testing it here https://ci.suse.de/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests-qe/7/

PR tests are unstable, impossible to test it there.
I'm gonna try in local.

**Comment by srbarrios** [comment]
I must have a wrong URL somewhere as I have this error:
```
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):           ID: proxy-container-packages
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):     Function: pkg.installed
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):       Result: False
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):      Comment: An error was encountered while installing package(s): Zypper command failure: Running scope as unit: run-rd1a86e063c3b4134a8ee2405d2bf36cb.scope
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):               Error building the cache:
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):               [proxy_client_tools_repo|http://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/openSUSE_Leap_15-Uyuni-Client-Tools/openSUSE_Leap_15.0/] Valid metadata not found at specified URL
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):               History:
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):                - Signature verification failed for repomd.xml
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):                - Can't provide /repodata/repomd.xml

module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):               Some of the repositories have not been refreshed because of an error.
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):               Package 'uyuni-proxy-systemd-services' not found.
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):      Started: 13:41:06.711947
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):     Duration: 545.829 ms
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec):      Changes:   
module.cucumber_testsuite.module.proxy.module.proxy.module.host.null_resource.provisioning[0] (remote-exec): ----------
```

**Comment by srbarrios** [comment]
I can confirm this is working for 4.3


```
oscar-pxy:/etc/zypp/repos.d # cat containers_pool_repo.repo
[containers_pool_repo]
enabled=1
autorefresh=1
baseurl=http://download.suse.de/ibs/SUSE/Products/SLE-Module-Containers/15-SP4/x86_64/product/
type=rpm-md
oscar-pxy:/etc/zypp/repos.d # cat containers_updates_repo.repo
[containers_updates_repo]
enabled=1
autorefresh=1
baseurl=http://download.suse.de/ibs/SUSE/Updates/SLE-Module-Containers/15-SP4/x86_64/update/
type=rpm-md
oscar-pxy:/etc/zypp/repos.d # cat /etc/os-release
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
```

File: salt/proxy/init.sls [hunk 1]
```diff
@@ -42,10 +42,10 @@ proxy-packages:
 
 {% if install_proxy_container_packages %}
 
-{% if 'head' in grains.get('product_version') %}
-    {% set client_tools_repo =  grains.get("mirror") | default("download.suse.de/ibs", true) ~ '/Devel:/Galaxy:/Manager:/Head:/SLE15-SUSE-Manager-Tools/images/repo/SLE-15-Manager-Tools-POOL-x86_64-Media1/' %}
-{% else %}
+{% if 'uyuni' in grains.get('product_version') %}
```
Comment (cbosdo) [inline salt/proxy/init.sls:45]: don't you need to handle `uyuni-pr` too here?

File: salt/server/testsuite.sls [hunk 2]
```diff
@@ -88,7 +88,7 @@ testsuite_salt_packages:
   pkg.installed:
     - pkgs:
       - salt-ssh
-{% if 'head' in grains.get('product_version') or 'uyuni-master' in grains.get('product_version') or 'nightly' in grains.get('product_version') %}
+{% if 'head' in grains.get('product_version') or 'uyuni-master' in grains.get('product_version') or 'nightly' in grains.get('product_version') or 'uyuni-pr' in grains.get('product_version') %}
```
Comment (srbarrios) [inline salt/server/testsuite.sls:91]: I'm not fully sure if I should add this case here.
It's out of this PR in fact. But looks like we missed in the past.

**Pull request merged**